badge-maker/lib/badge-renderers.js

```diff
@@ -87,46 +111,62 @@ function renderText({
 
   let renderedText = ''
   if (shadow) {
-    renderedText = `<text x="${x}" y="${shadowMargin}" fill="#010101" fill-opacity=".3" transform="scale(.1)" textLength="${outTextLength}">${escapedContent}</text>`
+    renderedText = `<text aria-hidden="true" x="${x}" y="${shadowMargin}" fill="#010101" fill-opacity=".3" transform="scale(.1)" textLength="${outTextLength}">${escapedContent}</text>`
   }
-  renderedText += `<text x="${x}" y="${textMargin}" transform="scale(.1)" textLength="${outTextLength}">${escapedContent}</text>`
+  renderedText += `<text x="${x}" y="${textMargin}" transform="scale(.1)" fill="#fff" textLength="${outTextLength}">${escapedContent}</text>`
 
   return {
-    renderedText,
+    renderedText: link
+      ? renderLink({
+          link,
+          height,
+          textLength,
+          horizPadding,
+          leftMargin,
+          renderedText,
+        })
+      : renderedText,
     width: textLength,
   }
 }
 
-function renderLinks({
-  links: [leftLink, rightLink] = [],
-  leftWidth,
-  rightWidth,
-  height,
-}) {
-  leftLink = escapeXml(leftLink)
-  rightLink = escapeXml(rightLink)
+function renderBadge(
+  {
+    links,
+    leftWidth,
+    rightWidth,
+    height,
+    label,
+    message,
+    // Override for aria if links are handled elsewhere (i.e social style)
+    hasLink,
+  },
+  main
+) {
+  const width = leftWidth + rightWidth
+
+  const leftLink = escapeXml(links[0])
+  const rightLink = escapeXml(links[1])
   const hasLeftLink = leftLink && leftLink.length
   const hasRightLink = rightLink && rightLink.length
-  const leftLinkWidth = hasRightLink ? leftWidth : leftWidth + rightWidth
 
-  function render({ link, width }) {
-    return `<a target="_blank" xlink:href="${link}"><rect width="${width}" height="${height}" fill="rgba(0,0,0,0)"/></a>`
-  }
+  const accessibleText = createAccessibleText({ label, message })
+  const excapedAccessibleText = escapeXml(accessibleText)
 
-  return (
-    (hasRightLink
-      ? render({ link: rightLink, width: leftWidth + rightWidth })
-      : '') +
-    (hasLeftLink ? render({ link: leftLink, width: leftLinkWidth }) : '')
-  )
-}
+  if (hasLink === undefined) {
+    hasLink = links && links.length && links.every(link => link)
```

Comment:
I'm not following the intent of `links.every(link => link)` here.
Are we trying to achieve something different from `hasLeftLink || hasRightLink` ?

Reply:
Yes, I will make this explicit.